### PR TITLE
{Telemetry} Fix #16579: Suppress telemetry exception in --debug when there is no valid account login

### LINKED ISSUE #16579: Telemetry reports issue when there is no valid account login
> ### `az feedback` auto-generates most of the information requested below, as of CLI version 2.0.62

**Describe the bug**

When there is no account login, although my command succeed. It will print the following exception to me when using `--debug`. Please catch the exception and not report to customer.

image

**To Reproduce**

**Expected behavior**

**Environment summary**

**Additional context**

### PR DESCRIPTION
**Pull request by PARADISSEEKR**
**Description**

Fix #16579 
**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

File: src/azure-cli-core/azure/cli/core/_profile.py [hunk 1]
```diff
@@ -494,32 +494,39 @@ def load_cached_subscriptions(self, all_clouds=False):
...
+                raise CLIError("Subscription '{}' not found. "
+                               "Check the spelling and casing and try again.".format(subscription))
+            if not result and not subscription:
+                raise CLIError("No subscription found. Run 'az account set' to select a subscription.")
+            if len(result) > 1:
+                raise CLIError("Multiple subscriptions with the name '{}' found. "
+                               "Specify the subscription ID.".format(subscription))
+            return result[0]
+        except:
+            return {}
```
Comment (jiasli) [inline src/azure-cli-core/azure/cli/core/_profile.py:523]: If `CLIError` is discarded, what is the point of raising it in the first place?

File: src/azure-cli-core/azure/cli/core/_profile.py [hunk 2]
```diff
@@ -494,32 +494,39 @@ def load_cached_subscriptions(self, all_clouds=False):
...
 
     def get_subscription_id(self, subscription=None):  # take id or name
-        return self.get_subscription(subscription)[_SUBSCRIPTION_ID]
+        try:
+            return self.get_subscription(subscription)[_SUBSCRIPTION_ID]
+        except :
+            return ''
```
Comment (jiasli) [inline src/azure-cli-core/azure/cli/core/_profile.py:529]: Suppressing exceptions in core is too aggressive. Please move the `except` logic to telemetry.

**Review by jiasli — CHANGES_REQUESTED** [review]

**Comment by fengzhou-msft** [comment]
@PARADISSEEKR Please catch the exceptions in `telemetry.py` for the purpose of fixing the mentioned issue, do not change error handling in `core/_profile.py`.

**Comment by fengzhou-msft** [comment]
Changed the title as it's not customer-facing. See [Format PR Title](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#format-pr-title).

File: src/azure-cli-core/azure/cli/core/decorators.py [hunk 3]
```diff
@@ -55,7 +55,7 @@ def hash256_result(func):
     @wraps(func)
     def _decorator(*args, **kwargs):
         val = func(*args, **kwargs)
-        if not val:
+        if val is None:
             raise ValueError('Return value is None')
```
Comment (fengzhou-msft) [inline src/azure-cli-core/azure/cli/core/decorators.py:59]: Can you check all the places that the `hash256_result` decorator is used and see the impact of this change?

Comment (PARADISSEEKR) [reply src/azure-cli-core/azure/cli/core/decorators.py:59]: All functions that use this decorator return strings, so there is no effect

**Review by fengzhou-msft — APPROVED** [review]

**Pull request merged**